src/servers/middlewares/auth-router.ts

```diff
@@ -59,35 +57,54 @@ authRouter.post("/browser", async (req, res) => {
 
 const jwtAlgorithms: jwt.Algorithm[] = ["HS256"]
 
-// used by oathkeeper to validate JWT
+// used by oathkeeper to validate LegacyJWT and SessionToken
 // should not be public
 authRouter.post(
   "/validatetoken",
   wrapAsyncToRunInSpan({
     namespace: "validatetoken",
-    fn: async (
-      /* eslint-disable @typescript-eslint/no-explicit-any */
-      req: Request<any, any, any, any, Record<string, any>>,
-      res: Response<any>,
-    ) => {
+    fn: async (req: express.Request, res: express.Response) => {
+      const accountsRepo = AccountsRepository()
       const headers = req?.headers
       let tokenPayload: string | jwt.JwtPayload | null = null
       const authz = headers.authorization || headers.Authorization
-      if (authz) {
-        try {
-          const rawToken = authz.slice(7) as string
-
-          tokenPayload = jwt.verify(rawToken, JWT_SECRET, {
-            algorithms: jwtAlgorithms,
-          })
-        } catch (err) {
-          res.status(401).send({ error: "Token validation error" })
+
+      if (!authz) {
+        res.status(401).send({ error: "Missing token" })
+        return
+      }
+
+      const rawToken = authz.slice(7) as string
+
+      // new flow
+      if (rawToken.length === 32) {
```

Comment:
The logic here regarding which token we have and how to validate / verify it should be its own use case - don't want this logic in the middleware IMO.

Reply:
it's meant to be just temporary in the transition stage. do you still think I should extract it?

Reply:
It would be cleaner but its not a must have if the conditional is temporary. Can leave it I guess.